### PR TITLE
Avoid attributeError crash in xonfig wizard

### PR DESCRIPTION
**Pull request by bobhy**
Fix for #3711

Looks like changes in #3704 didn't ripple outward to xonfig.py. It was using old field names.

Also https://github.com/xonsh/xonsh/blob/528d2f913ae40c89420fad02c665f3916aae44a0/xonsh/environ.py#L1867-L1877 would return unconverted `DefaultNotGiven` in `.doc_default` if called with environment variable not defined in `env._vars`.

### CODE REVIEW

**Comment by scopatz** [comment]
Thanks for putting this in @bobhy - It looks like there may be some legit bugs here

File: docs/customization.rst [hunk 1]
```diff
@@ -234,30 +234,4 @@ variables:
...
-The second option is to run xonsh in an Ansi-term buffer inside
-Emacs. This way you have to switch modes if you want do Emacs-style
-editing, but you keep xonsh's impressive completion.
-
-For this it is preferred to have xonsh installed with the
-prompt-toolkit. Then you can leave ``$SHELL_TYPE`` at its default.
-
-Emacs will prompt you for the path of the xonsh exeutable when you
-start up ``ansi-term``.
+see `emacs <editors.html>`_.
```
Comment (scopatz) [inline docs/customization.rst:237]: `See` or `Please see`

Comment (laloch) [reply docs/customization.rst:237]: How is it related to "attributeError crash in xonfig wizard"? This PR is in fact a 3-in-1 PR.

**Comment by bobhy** [comment]
Didn't mean to mix concerns.  Will repush with just the relevant fixes.

**Comment by bobhy** [comment]
Closing this PR, because I can't figure out how to remove the irrelevant other commits that got mixed int o this one.
Replacing with #3715.

**Pull request closed without merging**